import { catchError, tap } from 'rxjs/operators';
export function catchWithCount(selector) {
  return input$ => {
    let errors = [];
    const errorOp = catchError((err, caught) => {
      errors.push(err);
      return selector(err, errors.length, caught).pipe(errorOp);
    });
    return input$.pipe(tap(() => {
      errors = [];
    }), errorOp);
  };
}